configure.ac

```diff
@@ -225,8 +225,9 @@ AS_IF([test "x$with_systemdsystemunitdir" = "xyes" -o "x$with_systemdsystemunitd
      AS_IF([test "x$PKG_CONFIG" = "x"],AC_MSG_ERROR([systemd support requested but no pkg-config available to query systemd package]))
      def_systemdsystemunitdir=$($PKG_CONFIG --variable=systemdsystemunitdir systemd)
 
+     # TODO: This seems to fail when systemd is not installed so pkg-config can't find the variable, we don't error out there
```

Comment:
This comment is not really clear to me: are we currently not erroring out or should we not be?

Reply:
The comment can be removed as I tried to fix this with this PR.

In the case that `systemd` is not installed but `pkg-config` is:
- pkg-config will try to get the value here: https://github.com/DOMjudge/domjudge/pull/2220/files#diff-49473dca262eeab3b4a43002adb08b4db31020d190caaad1594b47f1d5daa810R226 but can't as systemd is not there, so not package specification can give the variable `systemdsystemunitdir`.
- We should fail, but we don't. I think because an empty string is stored and we don't fail on the non-zero exit code.

Next in the code:
`AS_IF([test "x$def_systemdsystemunitdir" = "x"]` will be true as the variable is empty. I copied the solution from here: https://github.com/DOMjudge/domjudge/pull/2220/files#diff-49473dca262eeab3b4a43002adb08b4db31020d190caaad1594b47f1d5daa810R224 but I'm not 100% if we can't rewrite this to make more sense, I think the auto is set here https://github.com/DOMjudge/domjudge/pull/2220/files#diff-49473dca262eeab3b4a43002adb08b4db31020d190caaad1594b47f1d5daa810R224 but I'm not sure where the case for `with_systemdsystemunitdir=yes` would be set.